bench/Autofac.Benchmarks/RootContainerResolveBenchmark.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+
+using Autofac;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Running;
+
+namespace Autofac.Benchmarks
+{
+    /// <summary>
+    /// Tests the performance of retrieving various simple components from a root container.
+    /// </summary>
+    public class RootContainerResolveBenchmark
+    {
+        private IContainer _container;
+
+        [Setup]
+        public void Setup()
+        {
+            var builder = new ContainerBuilder();
+            builder.RegisterType<Shared>().SingleInstance();
+            builder.RegisterType<NonSharedReflection>();
+            builder.Register(c => new NonSharedDelegate());
+            _container = builder.Build();
+        }
+
+        [Benchmark(Baseline = true)]
+        public void OperatorNew()
+        {
+            var instance = new NonSharedDelegate();
+            GC.KeepAlive(instance);
```

Comment:
Being unfamiliar with BenchmarkDotNet - is there a reason you keep them alive? Or is that so we don't get hit with GC overhead time after a method is executed?